### PR TITLE
Add Enhanced Monster AI

### LINKED ISSUE #191: Implement different enemy attacks
**User Story**
As a player character, I'd like enemies to have different attacks, so that I encounter different styles of enemies making the game more varied

**Acceptance Criteria**

- Enemies have other attacks other than just walking up to you and hitting you

**Notes**
Ideas:
- Ranged attack
- Suicide bomber (if they get to you before you kill then, you take damage - fighter would use a shield)
- Shoot in all directions
- Teleport
- Temporarily invisible
- Heal themselves/heal another monster
- Ghost that can fly through walls
- Poisonous (take damage over a range of turns)
- Paralyse you (you can't move/attack for a turn)

---

**Why is this feature needed? Please describe the problem your requested feature wants to solve**

All enemies have the same attack

**Describe the solution you'd like**

Enemies to have different attack styles

### LINKED ISSUE #215: Change enemy facing direction
**User Story**
As a dungeon runner, I'd like the enemy's to face me, so that they don't walk backwards or attack me backwards.

**Acceptance Criteria**

- The enemy sprites are either forward facing or face the player when they move about

### LINKED ISSUE #251: Unable to play 'Endless' game mode
**Bug Summary**
When attempting to start a game with the 'endless' game mode, the game crashes, rendering it unplayable.

**Test Case(s)**

---

**Expected behavior**
It should be possible to play the 'endless' game mode.

**Observed Behaviour**
It is not possible to play the 'endless' game mode.

**Steps To Reproduce**

1. Run the game
2. Click on 'endless mode'
3. Click on 'explore dungeon'
4. Observe the error message

**Environment**
- Version: fb6574066e692e10203379fae16661c531d43c1c
- OS: Arch Linux
- Browser: Firefox v75

---

**Code Examples**

**Stack Trace**

**Screenshots**

**Error Report**

`TypeError: payload is undefined`

`src/features/snackbar/reducer.js` @ line 50: ```itemReceived: `${payload.name}-${new Date().getTime()}` ```

---

**Additional context**

Seems to be an issue with the snackbar reducer.

### PR DESCRIPTION
**Pull request by Pyxxil**
GitHub Issue (If applicable): #215 , #191, #251 

## PR Type

What kind of change does this PR introduce?
- Feature

## What is the current behavior?
Monsters are all the same: They attack you when they get close. None of them have any variations in their attacks.

## What is the new behavior?
Some monsters have been changed to have differing AI's, such as healing themselves, attacking with ranged/magic, suicidal, etc.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] Tested code with current upstream repo
- [ ] Added new tests to the test suite for the feature
- [ ] Tested the full test suite, and ensured that the feature does not break current build.
- [ ] Docs have been added/updated which fit (for bug fixes / features)
- [x] Contains **NO** breaking changes
- [x] Associated with an issue (GitHub or internal)
- [ ] Checked licensing of Frameworks/Libraries (if applicable) 

## Other information
This is not yet completed, but I am putting it out there for:
a) feedback on what else should be added
b) awareness that we need to merge #221, #238, and #229 before this one as they contain breaking changes relating to this

Internal Issue (If applicable): closes #215, closes #191, closes #251

### CODE REVIEW

**Comment by Frosty273** [comment]
With #229 merged, I suggest the following AI for the added monsters when #238 is merged:

- Snake ~ normal or magical(poison dart)
- Salamander ~ normal or magical(fireball or poison dart)
- Water Spirit ~ magical(ice shard)
- Gas Cloud ~ suicidal
- Fungi Bulb ~ suicidal
- Fire Spirit ~ magical(fireball)
- Poison Cloud ~ suicidal or magical(poison cloud)
- Plantera ~ suicidal or magical(mend/heal)
- Pumpkin Ghost ~ magical(thunder cloud)

File: README.md [hunk 1]
```diff
@@ -12,15 +12,15 @@
...
-| ![check][checkmark] Character Creation            | ![check][checkmark] Customisable Outfit Colour        |                                           |
-| ![check][checkmark] Additional Items in Shop      | ![cross][cross] Diversity in Enemy Attacks            |                                           |
-| ![check][checkmark] Spellcasting System           |                                                       |                                           |
+| Definitely                                        | Hopefully                                      | Maybe                                    |
+| ------------------------------------------------- | ---------------------------------------------- | ---------------------------------------- |
+| ![check][checkmark] DnD Ability Scores for Player | ![check][checkmark] UI/UX Improvements         | ![cross][cross] Particle Effects         |
+| ![check][checkmark] DnD Classes and Races         | ![cross][checkmar] Tutorial                    | ![cross][cross] Class-specific Abilities |
```
Comment (Matteas-Eden) [inline README.md]: Typo here: `checkmar` vs `checkmark`

File: src/data/monsters/medusa/index.js [hunk 2]
```diff
@@ -1,4 +1,5 @@
 import MedusaSprite from './medusa.png';
+import MedusaFloppedSprite from './medusa-flopped.png';
```
Comment (Matteas-Eden) [inline src/data/monsters/medusa/index.js:2]: Typo here?

File: src/data/monsters/wolf/index.js [hunk 3]
```diff
@@ -1,4 +1,5 @@
 import WolfSprite from './wolf.png';
+import WolfFloppedSprite from './wolf-flopped.png';
```
Comment (Matteas-Eden) [inline src/data/monsters/wolf/index.js:2]: Is `Flopped` a typo or intended? Not sure while reading this.

**Review by Matteas-Eden — DISMISSED** [review]
Looks good to me, just some typos around the place. Most important one is the README.

File: src/data/monsters/wolf/index.js (hunk 3 above)
Comment (Pyxxil) [reply src/data/monsters/wolf/index.js:2]: https://imagemagick.org/script/command-line-options.php#flop

**Review by Matteas-Eden — APPROVED** [review]
Looks good to me!

**Pull request merged**